pkg/keystore/awsKeystore.go

```diff
@@ -40,8 +46,26 @@ func createAwsSession(config *AwsConfig) *session.Session {
 	return session.Must(session.NewSession())
 }
 
+func waitUntilValidSession(config *AwsConfig) *session.Session {
+	awsSession := createAwsSession(config)
+	if !config.RetryOnCredentialsError {
+		return awsSession
+	}
+
+	for {
+		_, err := awsSession.Config.Credentials.Get()
+		if err != credentials.ErrNoValidProvidersFoundInChain {
+			return awsSession
```

Comment:
Would it be better to return the error here? So that it's possible to debug why it failed.

Reply:
`ErrNoValidProvidersFoundInChain` is the one I'm expecting if it fails. maybe better to check 
```go
if err != nil {
```
?